### PR TITLE
[WIP] djvulibre: adopt

### PR DESCRIPTION
**Pull request by Anton-Latukha**
###### Motivation for this change

1. Old description said that there is a viewer - it is not here, in the `djview` package. But there is all other tools.
2. Old description said there is library in package - the fact there is none. Most document utils I am aware of ship with own copy of `djvulibre` and often statically include it, and most utils seems to depend on `djvulibre` binaries that are very unix-like flexible.

The value of package is all utils for DJVU management.
New description of the package much more on point, package has a set of binaries and mans for them.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/djvulibre/default.nix [hunk 1]
```diff
@@ -13,10 +13,10 @@ stdenv.mkDerivation rec {
   buildInputs = [ libjpeg libtiff librsvg libiconv ];
```
Comment (peterhoeg) [inline pkgs/applications/misc/djvulibre/default.nix]: While at it, please also split name into pname and version.

File: pkgs/applications/misc/djvulibre/default.nix [hunk 2]
```diff
@@ -13,10 +13,10 @@ stdenv.mkDerivation rec {
...
 
   meta = with stdenv.lib; {
-    description = "A library and viewer for the DJVU file format for scanned images";
+    description = "The big set of CLI tools to make/modify/optimize/show/export DJVU files";
     homepage = http://djvu.sourceforge.net;
```
Comment (peterhoeg) [inline pkgs/applications/misc/djvulibre/default.nix]: Quote the url.

File: pkgs/applications/misc/djvulibre/default.nix [hunk 3]
```diff
@@ -13,10 +13,10 @@ stdenv.mkDerivation rec {
   buildInputs = [ libjpeg libtiff librsvg libiconv ];
 
   meta = with stdenv.lib; {
-    description = "A library and viewer for the DJVU file format for scanned images";
+    description = "The big set of CLI tools to make/modify/optimize/show/export DJVU files";
```
Comment (peterhoeg) [inline pkgs/applications/misc/djvulibre/default.nix]: Nitpick - I don't know if this is really "The big set" but it's definitely "A set".

**Review by peterhoeg — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/djvulibre/default.nix (hunk 3 above)
Comment (peterhoeg) [reply pkgs/applications/misc/djvulibre/default.nix]: By the way, it's great that you're adopting this!!

Comment (Anton-Latukha) [reply pkgs/applications/misc/djvulibre/default.nix]: (Wanted to underline that all needed tools are indeed in the package, if person up for it. & Thank god set is not _large_.)

I love to gradually collect beloved & unique packages, who I found astray, and bring them to well-earned citizenship, like `Handbrake` (I know much about theory and practice of encoding), `qbittorrent`, `wmctrl` (draw your attention - this one is a real treasure for X11 desktops).

I love to adopt & maintain the old unique userspace-y packages, what to not love about it.
If I commit to a package as a maintainer - I indeed become a maintainer, update to major versions, control the PRs that people send into, keeping the logic of code. Current low frequency thou allows me to forget the recommendations, please sorry for that, I would improve.

File: pkgs/applications/misc/djvulibre/default.nix (hunk 2 above)
Comment (Anton-Latukha) [reply pkgs/applications/misc/djvulibre/default.nix]: Oh, thanks, you saying that to me for 2 or 3 time. So..., I am writing write recommendation to myself.

File: pkgs/applications/misc/djvulibre/default.nix (hunk 1 above)
Comment (Anton-Latukha) [reply pkgs/applications/misc/djvulibre/default.nix]: I wanted to do code changes after adopting a package, to be "by maintainer".
Done.

File: pkgs/applications/misc/djvulibre/default.nix (hunk 2 above)
Comment (Anton-Latukha) [reply pkgs/applications/misc/djvulibre/default.nix]: Done.

**Comment by Anton-Latukha** [comment]
What the heck.
First time bot tagget it being no rebuild.
Probably it is due to changes in upstream URL.
I don't think that `homepage` change triggers recompilation.

---

I see:
```fish
pkgs/top-level/all-packages.nix
pkgs/applications/kde/okular.nix
pkgs/tools/typesetting/pdf2djvu/default.nix
pkgs/tools/typesetting/djvu2pdf/default.nix
pkgs/tools/graphics/pdfread/default.nix
pkgs/applications/search/recoll/default.nix
pkgs/applications/graphics/djview/default.nix
pkgs/applications/graphics/gscan2pdf/default.nix
pkgs/applications/misc/djvulibre/default.nix
pkgs/applications/misc/gImageReader/default.nix
pkgs/applications/graphics/ImageMagick/7.0.nix
pkgs/applications/graphics/ImageMagick/default.nix
pkgs/applications/misc/qpdfview/default.nix
pkgs/applications/misc/k2pdfopt/default.nix
pkgs/applications/misc/zathura/djvu/default.nix
pkgs/desktops/gnome-3/core/evince/default.nix
```

I see, `ImageMagick` is probably _the_ big tree dependency.

**Comment by Anton-Latukha** [comment]
But the truth is:
```fish
> nix-store -q --tree .
---
/nix/store/dsxy8jk6iahbwmpr23pfp0qyrfgqkj03-djvulibre-3.5.27-bin
+---/nix/store/6yaj6n8l925xxfbcd65gzqx3dz7idrnn-glibc-2.27
|   +---/nix/store/6yaj6n8l925xxfbcd65gzqx3dz7idrnn-glibc-2.27 [...]
+---/nix/store/najvrr9mppdrgjh529l79w8px36wn58z-gcc-8.3.0-lib
|   +---/nix/store/6yaj6n8l925xxfbcd65gzqx3dz7idrnn-glibc-2.27 [...]
|   +---/nix/store/najvrr9mppdrgjh529l79w8px36wn58z-gcc-8.3.0-lib [...]
+---/nix/store/fvbq2jxyib3gm5vfgqf6yylz4mkg3qfr-djvulibre-3.5.27
|   +---/nix/store/6yaj6n8l925xxfbcd65gzqx3dz7idrnn-glibc-2.27 [...]
|   +---/nix/store/nair5j8i8p9ik286ijkhyaip9nychsvy-libjpeg-turbo-2.0.3
|   |   +---/nix/store/6yaj6n8l925xxfbcd65gzqx3dz7idrnn-glibc-2.27 [...]
|   +---/nix/store/najvrr9mppdrgjh529l79w8px36wn58z-gcc-8.3.0-lib [...]
|   +---/nix/store/fvbq2jxyib3gm5vfgqf6yylz4mkg3qfr-djvulibre-3.5.27 [...]
+---/nix/store/l6h4ya0wzb4b8mr0y58k2gh2nhfql4sn-bash-4.4-p23
|   +---/nix/store/6yaj6n8l925xxfbcd65gzqx3dz7idrnn-glibc-2.27 [...]
|   +---/nix/store/l6h4ya0wzb4b8mr0y58k2gh2nhfql4sn-bash-4.4-p23 [...]
+---/nix/store/wmlvk1gim9plg5719i02zykyajlka72y-libtiff-4.0.10
    +---/nix/store/6yaj6n8l925xxfbcd65gzqx3dz7idrnn-glibc-2.27 [...]
    +---/nix/store/62ar9xmrlcnlgmwgfi77xz6bq1180vhi-zlib-1.2.11
    |   +---/nix/store/6yaj6n8l925xxfbcd65gzqx3dz7idrnn-glibc-2.27 [...]
    +---/nix/store/nair5j8i8p9ik286ijkhyaip9nychsvy-libjpeg-turbo-2.0.3 [...]
    +---/nix/store/najvrr9mppdrgjh529l79w8px36wn58z-gcc-8.3.0-lib [...]
    +---/nix/store/yk8lp0x7sml42gaf9gfkr6cwx99glz7b-xz-5.2.4
    |   +---/nix/store/6yaj6n8l925xxfbcd65gzqx3dz7idrnn-glibc-2.27 [...]
    |   +---/nix/store/yk8lp0x7sml42gaf9gfkr6cwx99glz7b-xz-5.2.4 [...]
    +---/nix/store/wmlvk1gim9plg5719i02zykyajlka72y-libtiff-4.0.10 [...]
```

**Comment by Anton-Latukha** [comment]
@peterhoeg

What to do in the case, `djvulibre` clame to fame is the main library, package produces only binaries. I tried `lib` out, there is none. All dependencies work.
Should I further seek to add library, maybe split tools & lib, should we go and force tools to use the NixOS lib.
Or maybe it does not worth this effort.

**Comment by peterhoeg** [comment]
Ideally this should indeed be split into a lib and a set of tools. Considering the rebuilding, we should target staging instead of master so if you have the time to do that split, we might as well do that first.

**Comment by Anton-Latukha** [comment]
Moving to WIP.

`INFO:` Looked-up, GitHub currently does not released a feature to move PR to the draft state.

So marking `[WIP]`.

**Comment by Anton-Latukha** [comment]
I love thou the small atomic PRs. This one becomes not such. I have a painful expirience with my epic PRs not ever get merged.

**Comment by Anton-Latukha** [comment]
Woops.

**Pull request closed without merging**